docs/manual/src/interface/functions.md

```diff
@@ -0,0 +1,70 @@
+# Functions
+
+The interface definition can contain public functions using the standard Rust syntax:
+
+```rust
+#[uniffi::declare_interface]
+mod example {
+    fn hello_world() -> String {
+        "Hello World!".to_owned()
+    }
+}
+```
+
+The arguments and return types must be types that are understood by UniFFI.
+
+UniFFI does not understand path references or type aliases, so things like
+the following will produce a compile-time error:
+
+```rust
+
+struct NotPartOfTheInterface {
+    value: u32,
+}
+
+#[uniffi::declare_interface]
+mod example {
+
+    type MyString = String
+
+    // Error: UniFFI doesn't know what "MyString" is.
+    fn hello_world() -> MyString {
+        "Hello World!".to_owned()
+    }
+
+    // Error: UniFFI doesn't know about the `NotPartOfTheInterface` type.
+    fn example(v: NotPartOfTheInterface) {
+        println!("value: {}", v.value);
+    }
+
+    // Error: UniFFI doesn't understand the `std::vec::` path; just use `Vec<T>`.
```

Comment:
Yeah, it's tricky. The Rust language has so much complexity/power that I think it's hard to inhabit a middle ground where you parse the Rust code independently of the compiler and correctly infer its semantics. In order to be safe, you need to either:

* Let the Rust compiler actually compile the code and output the information you want as a side-effect. This is how `wasm-bindgen` works for example, it makes Rust actually build the code and then arranges for the resulting library to have an extra data section with details about the types it encountered.
* Parse a subset of Rust that is so restricted that you can't possibly misinterpret what it means. That's what I was going for here. In fact, I think if we want to do this for real, we might even want to forcibly inject e.g. a `use std::vec::Vec` into the code so that we know that when we see a type named `Vec`, it is 100% definitely the `Vec` from the stdlib and some smart-alec hasn't done `use std::option::Option as Vec` at the top of the file for a laugh.